web/server/routers/post.ts

```diff
@@ -147,16 +148,16 @@ export const postRouter = router({
         }
       } catch (e) {
         throw new TRPCError({
-          message: "Server unable to access user endpoint.",
+          message: "An unexpected error occured.",
           code: "INTERNAL_SERVER_ERROR",
         });
       }
       try {
-        const post = await getPost(input.postOid);
-        const email = fosterTypeEmails[post.type as FosterType] || "";
+        const post = (await getPost(input.postOid)) as IPost;
```

Comment:
Modify `getPost` database action return type instead of using type assertion to get the type inference: `Promise<IPost & { _id: string; __v: number }>`